jboss-kie-workbench/added/launch/jboss-kie-workbench.sh

```diff
@@ -164,48 +170,103 @@ function configure_ha() {
-                JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dappformer-jms-username=${jmsBrokerUsername}"
-                JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dappformer-jms-password=${APPFORMER_JMS_BROKER_PASSWORD}"
-                JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Des.set.netty.runtime.available.processors=false"
-                JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.appformer.ext.metadata.index=elastic"
-                JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.appformer.ext.metadata.elastic.host=${APPFORMER_ELASTIC_HOST}"
-                JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.appformer.ext.metadata.elastic.port=${APPFORMER_ELASTIC_PORT:-9300}"
-                JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.appformer.ext.metadata.elastic.cluster=${APPFORMER_ELASTIC_CLUSTER_NAME:-kie-cluster}"
-                JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.appformer.ext.metadata.elastic.retries=${APPFORMER_ELASTIC_RETRIES:-10}"
-
-                # [RHPAM-1522] make the workbench webapp distributable for HA (2 steps)
-                local web_xml="${JBOSS_HOME}/standalone/deployments/ROOT.war/WEB-INF/web.xml"
-                sed -i "/^\s*<!--/!b;N;/<distributable\/>/s/.*\n//;T;:a;n;/^\s*-->/!ba;d" "${web_xml}"
-                # step 2) modify the web cache container per https://access.redhat.com/solutions/2776221
-                #         note: the below differs from the EAP 7.1 solution above, since EAP 7.2
-                #               doesn't have "mode", "l1", and "owners" attributes in the original config
-                local web_cache="\
-                    <transport lock-timeout='60000'/>\
-                    <replicated-cache name='repl'>\
-                        <file-store/>\
-                    </replicated-cache>\
-                    <distributed-cache name='dist'>\
-                        <file-store/>\
-                    </distributed-cache>"
-                xmllint --shell "${JBOSS_HOME}/standalone/configuration/standalone-openshift.xml" << SHELL
-                    cd //*[local-name()='cache-container'][@name='web']
-                    set ${web_cache}
-                    save
-SHELL
-# SHELL line above not indented on purpose for correct vim syntax highlighting
+            if [ -n "$jmsBrokerUsername" -a -n "$APPFORMER_JMS_BROKER_PASSWORD" -a -n "$APPFORMER_JMS_BROKER_ADDRESS" ] ; then
+                if [ -n "$APPFORMER_INFINISPAN_SERVICE_NAME" -o -n "$APPFORMER_INFINISPAN_HOST" -o -n "$APPFORMER_ELASTIC_SERVICE_NAME"  -o -n "$APPFORMER_ELASTIC_HOST" ] ; then
+                    # set the workbench properties for HA
+                    configure_ha_common
+                    if [ -n "$APPFORMER_INFINISPAN_SERVICE_NAME" -o -n "$APPFORMER_INFINISPAN_HOST" ] ; then
+                        configure_ha_infinispan
+                    else
+                        configure_ha_elastic
```

Comment:
Shouldn't we double check if the ES envs are present before configure it instead to defaulting to ES if JDG is not correctly set?

Reply:
@spolti : They were checked above already. The only reason that infinspan needed another check was to differentiate. However, I recognize this wasn't clear, so I've cleaned it up and pushed again. Please have another look. Thanks!